### PR TITLE
aspire: Remove support for legacy resources

### PR DESCRIPTION
**Pull request by ellismg**
During early previews of Aspire, there was a large set of custom resource types that Aspire and `azd` supported. As the product evolved, these types were removed from the manifest in favor of a single `azure.bicep.v0` type, but we retained our support for the older stuff as a point in time thing.

That point in time has now passed, Aspire has GA'd without any of these resource types in it. We can now remove our support for them and remove a bunch of logically dead code.

### CODE REVIEW

**Comment by ellismg** [comment]
@vhvb1989 - I don't want to cause any rebase pain for you with the stuff you have in flight, so I'll take care in landing this after your stuff goes in, but I wanted to get the ball rolling on starting to clean up a bunch of what I think is dead code from early aspire previews where we were not on the CDK plan.

**Comment by davidfowl** [comment]
Clean!

File: cli/azd/pkg/apphost/generate.go [hunk 1]
```diff
@@ -1570,7 +1409,6 @@ func (b *infraGenerator) buildArgsBlock(args []string, manifestCtx *genContainer
 		// This logic is similar to what we do in buildEnvBlock to detect when we need to take values and treat them as ACA
 		// secrets.
 		if strings.Contains(arg, ".connectionString}") ||
-			strings.Contains(resolvedArg, "{{ connectionString") ||
```
Comment (vhvb1989) [inline cli/azd/pkg/apphost/generate.go:1573]: container.v0 has `connectionString` (https://github.com/dotnet/aspire/blob/main/src/Schema/aspire-8.0.json#L64)

Comment (ellismg) [reply cli/azd/pkg/apphost/generate.go:1788]: This case should be handled today.  This is removing the support for the `connectionString` function which was supported in our container app templates and only supported a subset of resources (the old `azure.*` onces, as well as postgres and redis for the ACA dev services) and none of these types of resources are supported anymore, so we will never emit calls to this helper.  I have also deleted the helper method as part of this PR.

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**